schema/testdata.go

```diff
@@ -1,131 +1,242 @@
 package schema
 
 import (
-	"net"
+	"fmt"
+	"reflect"
 	"sort"
+	"strconv"
 	"strings"
 	"time"
 
 	"github.com/apache/arrow/go/v13/arrow"
 	"github.com/apache/arrow/go/v13/arrow/array"
 	"github.com/apache/arrow/go/v13/arrow/memory"
+	"github.com/cloudquery/plugin-sdk/v2/schema"
 	"github.com/cloudquery/plugin-sdk/v3/types"
 	"github.com/google/uuid"
 )
 
-func TestSourceTable(name string) *Table {
-	return &Table{
-		Name:        name,
-		Description: "Test table",
-		Columns: ColumnList{
-			CqIDColumn,
-			CqParentIDColumn,
-			{
-				Name:            "uuid_pk",
-				Type:            types.ExtensionTypes.UUID,
-				CreationOptions: ColumnCreationOptions{PrimaryKey: true},
-			},
-			{
-				Name:            "string_pk",
-				Type:            arrow.BinaryTypes.String,
-				CreationOptions: ColumnCreationOptions{PrimaryKey: true},
-			},
-			{
-				Name: "bool",
-				Type: arrow.FixedWidthTypes.Boolean,
-			},
-			{
-				Name: "int",
-				Type: arrow.PrimitiveTypes.Int64,
-			},
-			{
-				Name: "float",
-				Type: arrow.PrimitiveTypes.Float64,
-			},
-			{
-				Name: "uuid",
-				Type: types.ExtensionTypes.UUID,
-			},
-			{
-				Name: "text",
-				Type: arrow.BinaryTypes.String,
-			},
-			{
-				Name: "text_with_null",
-				Type: arrow.BinaryTypes.String,
-			},
-			{
-				Name: "bytea",
-				Type: arrow.BinaryTypes.Binary,
-			},
-			{
-				Name: "text_array",
-				Type: arrow.ListOf(arrow.BinaryTypes.String),
-			},
-			{
-				Name: "text_array_with_null",
-				Type: arrow.ListOf(arrow.BinaryTypes.String),
-			},
-			{
-				Name: "int_array",
-				Type: arrow.ListOf(arrow.PrimitiveTypes.Int64),
-			},
-			{
-				Name: "timestamp",
-				Type: arrow.FixedWidthTypes.Timestamp_us,
-			},
-			{
-				Name: "json",
-				Type: types.ExtensionTypes.JSON,
-			},
-			{
-				Name: "uuid_array",
-				Type: arrow.ListOf(types.ExtensionTypes.UUID),
-			},
-			{
-				Name: "inet",
-				Type: types.ExtensionTypes.Inet,
-			},
-			{
-				Name: "inet_array",
-				Type: arrow.ListOf(types.ExtensionTypes.Inet),
-			},
-			{
-				Name: "cidr",
-				Type: types.ExtensionTypes.Inet,
-			},
-			{
-				Name: "cidr_array",
-				Type: arrow.ListOf(types.ExtensionTypes.Inet),
-			},
-			{
-				Name: "macaddr",
-				Type: types.ExtensionTypes.Mac,
-			},
-			{
-				Name: "macaddr_array",
-				Type: arrow.ListOf(types.ExtensionTypes.Mac),
-			},
-		},
+// TestSourceOptions controls which types are included by TestSourceColumns.
+type TestSourceOptions struct {
```

Comment:
What about adding a shorthand to set all options to `true`? Something like `func AllTestSourceOptions() TestSourceOptions` and then every field set to true.

Reply:
I'd even swap for `SkipTypeX` semantics, so that the default would be everything and if not, the destination has to be mindful about it (just like of all the skip modes for writing)

Reply:
seems that this testing suite is growing I would do two things, one is more important then the other imo:

1) I would use `Skip` - this way everything is included by default as usually destinations should support everything, unlike modes of operation the SDK doesn't error out (at least for now) if a mode is not supported so if a destinations updates to a newer SDK with new arrow types this would error out in "prod" instead of tests.
2) More of a preference - I think mostly we used `WithOption()` pattern so I would stick to that but that's not a must.